firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java

```diff
@@ -4548,6 +4548,57 @@ private static FirebaseApp appForDatabaseUrl(String url, String name) {
         name);
   }
 
+  @Test
+  public void testGetPropagatesEventsToListeners() // TODO(mtewani): Convert into smoke test
+      throws DatabaseException, InterruptedException, ExecutionException {
+    /**
+     * There's a situation where if you have a pending get() and then an onValue() at the same
+     * location and get resolves before onValue does, onValue gets no listeners
+     */
+    int counter = 0;
+    boolean pendingGet = false;
+    while (counter++ < 5 || !pendingGet) {
+      FirebaseApp writeApp =
+          appForDatabaseUrl(IntegrationTestValues.getDatabaseUrl(), UUID.randomUUID().toString());
+      // To ensure that we don't read the cached results, we need a separate app.
+      FirebaseApp readApp =
+          appForDatabaseUrl(IntegrationTestValues.getDatabaseUrl(), UUID.randomUUID().toString());
+      FirebaseDatabase readDb = FirebaseDatabase.getInstance(readApp);
+      FirebaseDatabase writeDb = FirebaseDatabase.getInstance(writeApp);
+      DatabaseReference writeNode = writeDb.getReference().push();
+      DatabaseReference readNode = IntegrationTestHelpers.translateReference(writeNode, readDb);
+      try {
+        Map<String, Object> toWrite =
+            new MapBuilder().put("foo1", "child1").put("foo2", "child2").build();
+        await(writeNode.setValue(toWrite));
+        Task<DataSnapshot> getTask = readNode.get();
+        AtomicBoolean receivedGetValue = new AtomicBoolean();
+        getTask.addOnCompleteListener(
+            new OnCompleteListener<DataSnapshot>() {
+              @Override
+              public void onComplete(@NonNull Task<DataSnapshot> task) {
+                DataSnapshot snapshot = task.getResult();
+                assertEquals(toWrite, snapshot.getValue());
+                receivedGetValue.set(true);
+              }
+            });
+        pendingGet = !receivedGetValue.get();
+        DatabaseReference childRef = readNode.child("foo1");
+        new ReadFuture(
+                childRef,
+                events -> {
```

Comment:
I think this callback will run on the platform event loop, but where does that `getTask.addOnCompleteListener(...)` callback run? I think we have to pass the runloop explicitly.

Reply:
https://github.com/firebase/firebase-android-sdk/blob/master/firebase-database/src/main/java/com/google/firebase/database/core/Repo.java#L540

Reply:
That `getTask` is just chaining off of the one in `getValue`, so I thought they would be using the same thread...And `setValue`'s test doesn't seem to specify https://github.com/firebase/firebase-android-sdk/blob/fbbe3e20aaa8d1ef4c4740a02ef8237329ca14a6/firebase-database/src/androidTest/java/com/google/firebase/database/QueryTest.java#L1592

Reply:
Ah, you're right.